libs/angular-yandex-maps-v3/src/lib/components/common/y-map-hint/y-map-hint.directive.ts

```diff
@@ -0,0 +1,189 @@
+ *
+ * In a nutshell, each entity (YMapFeature, YMapMarker, YMapHotspot) has a `properties` property, which is `Record<string, unknown>`.
+ * When you define these components, you can pass any metadata to this property, which you can later access using `YMapHintProps['hint']`.
+ * When you pass a callback to `YMapHintProps['hint']`, you can read the `properties` property and return the hint based on the metadata.
+ * What you return from `YMapHintProps['hint']` will be available in `<ng-template />` (it's set via $implicit, so you can use any name for let-*).
+ *
+ * ```html
+ * <y-map
+ *   [props]="{
+ *     location: {
+ *       center: [-0.127696, 51.507351],
+ *       zoom: 9,
+ *     },
+ *   }"
+ * >
+ *   <y-map-default-scheme-layer />
+ *   <y-map-default-features-layer />
+ *
+ *   <!-- onHint = (object?: YMapFeature | YMapMarker | YMapHotspot) => object?.properties?.['anyNameKey']; -->
+ *   <!-- It accepts an entity on which a user hovered and returns a context, which is set to $implicit in ng-template. -->
+ *   <!-- Your function can return anything since we have access to the returned value. -->
+ *   <y-map-hint [props]="{ hint: onHint }">
+ *     <ng-template let-context>
+ *       <div
+ *         style="
+ *           background-color: #fff;
+ *           border-radius: 6px;
+ *           padding: 3px 6px 4px;
+ *           margin-left: 15px;
+ *         "
+ *       >
+ *         <!-- context.hint is the return value of onHint -->
+ *         {{ context?.hint }}
+ *       </div>
+ *     </ng-template>
+ *   </y-map-hint>
+ *
+ *   <y-map-default-marker
+ *     [props]="{
+ *       coordinates: [-0.127696, 51.507351],
+ *       properties: {
+ *         anyNameKey: 'Central London',
+ *       },
+ *     }"
+ *   />
+ * </y-map>
+ * ```
+ */
+@Directive({
+  selector: 'y-map-hint',
+  standalone: true,
+})
+export class YMapHintDirective implements AfterContentInit, OnChanges, OnDestroy {
+  @ContentChild(TemplateRef) template?: TemplateRef<any>;
+
+  private readonly destroy$ = new Subject<void>();
+
+  private hint?: YMapHint;
+
+  /**
+   * See the API entity documentation for detailed information. Supports ngOnChanges.
+   * {@link https://yandex.ru/dev/jsapi30/doc/ru/ref/packages/hint/#YMapHintProps}
+   */
+  @Input({ required: true }) props!: YMapHintProps;
+
+  /**
+   * The entity instance is created. This event runs outside an Angular zone.
+   */
+  @Output() ready: EventEmitter<YReadyEvent<YMapHint>> = new EventEmitter<YReadyEvent<YMapHint>>();
+
+  constructor(private readonly yMapComponent: YMapComponent) {}
+
+  ngAfterContentInit() {
+    this.yMapComponent.map$
+      .pipe(
+        filter(Boolean),
+        switchMap((map) =>
+          // It's safe to call it each time, the Yandex.Maps API handles multiple requests under the hood.
+          from(ymaps3.import('@yandex/ymaps3-hint@0.0.1')).pipe(
+            tap(({ YMapHint, YMapHintContext }) => {
```

Comment:
`YMapHintContext` is undefined, we need to adjust `import`, so it can return an object, not just a constructor. And probably we need to mock an import inside a test, not in the utility. Otherwise, where should we mock an object with two instances?